File: server/src/models/Address.js
import mongoose from 'mongoose';

const addressSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  fullName: {
    type: String,
    required: true,
    trim: true
  },
  mobileNumber: {
    type: String,
    required: true,
    trim: true,
    validate: {
      validator: function(v) {
        return /^\d{10}$/.test(v);
      },
      message: 'Mobile number must be 10 digits'
    }
  },
  pincode: {
    type: String,
    required: true,
    trim: true,
    validate: {
      validator: function(v) {
        return /^\d{6}$/.test(v);
      },
      message: 'Pincode must be 6 digits'
    }
  },
  state: {
    type: String,
    required: true,
    trim: true
  },
  address: {
    type: String,
    required: true,
    trim: true
  },
  flatHouseCompany: {
    type: String,
    required: true,
    trim: true
  },
  city: {
    type: String,
    required: true,
    trim: true
  },
  landmark: {
    type: String,
    trim: true,
    default: ''
  },
  isDefault: {
    type: Boolean,
    default: false
  },
  addressType: {
    type: String,
    enum: ['Home', 'Office', 'Other'],
    default: 'Home'
  }
}, {
  timestamps: true
});

// Ensure only one default address per user
addressSchema.index({ userId: 1, isDefault: 1 }, { unique: true, partialFilterExpression: { isDefault: true } });

export default mongoose.model('Address', addressSchema);

File: jc_timber_mobile/lib/models/address.dart
/// Address model - matches backend Address schema.
class Address {
  final String id;
  final String fullName;
  final String mobileNumber;
  final String pincode;
  final String state;
  final String address;
  final String flatHouseCompany;
  final String city;
  final String landmark;
  final bool isDefault;
  final String addressType;

  Address({
    required this.id,
    required this.fullName,
    required this.mobileNumber,
    required this.pincode,
    required this.state,
    required this.address,
    required this.flatHouseCompany,
    required this.city,
    this.landmark = '',
    this.isDefault = false,
    this.addressType = 'Home',
  });

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      id: json['id']?.toString() ?? json['_id']?.toString() ?? '',
      fullName: json['fullName'] ?? '',
      mobileNumber: json['mobileNumber'] ?? '',
      pincode: json['pincode'] ?? '',
      state: json['state'] ?? '',
      address: json['address'] ?? '',
      flatHouseCompany: json['flatHouseCompany'] ?? '',
      city: json['city'] ?? '',
      landmark: json['landmark'] ?? '',
      isDefault: json['isDefault'] == true,
      addressType: json['addressType'] ?? 'Home',
    );
  }

  Map<String, dynamic> toJson() => {
        'fullName': fullName,
        'mobileNumber': mobileNumber,
        'pincode': pincode,
        'state': state,
        'address': address,
        'flatHouseCompany': flatHouseCompany,
        'city': city,
        'landmark': landmark,
        'isDefault': isDefault,
        'addressType': addressType,
      };
}
